envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/CallWithOverriddenHostHeaderTest.kt

```diff
@@ -0,0 +1,35 @@
+package pl.allegro.tech.servicemesh.envoycontrol
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlRunnerTestApp
+import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlTestConfiguration
+import pl.allegro.tech.servicemesh.envoycontrol.config.containers.HttpBinContainer
+
+internal class CallWithOverriddenHostHeaderTest : EnvoyControlTestConfiguration() {
+
+    companion object {
+        @JvmStatic
+        @BeforeAll
+        fun setupTest() {
+            setup(appFactoryForEc1 = { consulPort ->
+                EnvoyControlRunnerTestApp(consulPort = consulPort)
+            })
+        }
+    }
+
+    @Test
+    fun `should override Host header with value from specified custom header`() {
```

Comment:
IMO we should check in a test, that our custom header is not propagated to upstream. 
It will be a problem if this header is propagated to next services in the chain, because there is a risk that the next service will pass it to another, effectively changing the host header for whole service chain.

If the header is propagated, I am afraid we will have to implement a service whitelist or enable it on Envoy metadata level as @lukidzi proposed.

Reply:
I have checked and this header is propagated to upstream. So, after all, we need this whitelist or enabler in Envoy metadata @lukidzi @pbetkier @pzmi

Reply:
added possibility to enable in metadata per route